Adding a page for Aikar's tunings to PaperDocs (#129)
PaperMC/PaperDocs: source/server/aikar-tuning.rst

===============
Aikar's Flags
===============

.. contents::
   :depth: 1
   :local:

> Recommended JVM Startup Flags
> =================================================================

Reviewer: ```suggestion
Recommended JVM Startup Flags
=============================
```

===============

> Aikar's Tuning

Reviewer: ```suggestion
Aikar's Flags
```

Same for the file name, this is the much more common name and will be more recognizable.

===============

.. contents::
   :depth: 1
   :local:

> The JVM Startup Flags to use – MC 1.15+ (Java 8+, MC 1.8+) Update

Reviewer: ```suggestion
Recommended JVM Startup Flags
```

This section header seems a bit off for a docs article, something like that would be better.

=================================================================

> **Use these flags exactly, only changing Xmx and Xms. These flags work and scale accordingly to any size of memory, even 500MB but 1.15+ will not do well with such low memory…)**

Reviewer: ```suggestion
**Use these flags exactly, only changing Xmx and Xms. These flags work and scale accordingly to any size of memory, even 500MB but modern Minecraft versions will not do well with such low memory…)**
```

I'd like to avoid Minecraft version-specific stuff as much as possible, as it just leads to people asking "Will this work for 1.19 etc."

.. code-block:: bash

   java -Xms10G -Xmx10G -XX:+UseG1GC -XX:+ParallelRefProcEnabled -XX:MaxGCPauseMillis=200 
   -XX:+UnlockExperimentalVMOptions -XX:+DisableExplicitGC -XX:+AlwaysPreTouch 
   -XX:G1NewSizePercent=30 -XX:G1MaxNewSizePercent=40 -XX:G1HeapRegionSize=8M 
   -XX:G1ReservePercent=20 -XX:G1HeapWastePercent=5 -XX:G1MixedGCCountTarget=4 
   -XX:InitiatingHeapOccupancyPercent=15 -XX:G1MixedGCLiveThresholdPercent=90 
   -XX:G1RSetUpdatingPauseTimePercent=5 -XX:SurvivorRatio=32 -XX:+PerfDisableSharedMem 
   -XX:MaxTenuringThreshold=1 -Dusing.aikars.flags=https://mcflags.emc.gs 
   -Daikars.new.flags=true -jar paperclip.jar nogui

> LAST UPDATED: **Apr 25 2020 3:30PM EST**    
> 
> These flags are recommended for ALL versions of Minecraft! 1.8 all the way to 1.15+, use this set.

Reviewer: ```suggestion
```
See above, will lead to more questions.

.. warning::
   **IMPORTANT – READ – Don’t use ALL of your memory!! PTERODACTYL USERS!**

   When setting the Xms and Xmx values, if your host says you have 8000M memory, DO NOT USE 8000M! Minecraft (and Java) needs additional memory on top of that Xmx parameter. 
   It is recommended to reduce your Xmx/Xms by about **1000-1500M** to avoid running out of memory or “OOMKiller” hitting your server. 
   This also leaves room for the Operating System to use memory too.
   **Have 8000M memory? Use 6500M for safety.** *But you may also ask your host if they will cover this overhead for you and give you 9500M instead. Some hosts will! Just ask.*

Recommended Memory
==================

**We recommend using at least 6-10GB**, no matter how few players! If you can’t afford 10GB of memory, give as much as you can, 
but ensure you leave the operating system some memory too. G1GC operates better with more memory.

> 

Reviewer: Something here like "more memory ≠ more performance above a certain point", with the intent of preventing people from going out and buying 24gb from some shared host where that'll give them almost no benefit and just have them paying more.

Java GC Logging
===============

Are you having old gen issues with these flags? Add the following flags based on your java version to enable GC Logging:

**Java 8-10**

.. code-block:: bash

   -Xloggc:gc.log -verbose:gc -XX:+PrintGCDetails -XX:+PrintGCDateStamps -XX:+PrintGCTimeStamps 
   -XX:+UseGCLogFileRotation -XX:NumberOfGCLogFiles=5 -XX:GCLogFileSize=1M

**Java 11+**

.. code-block:: bash

   -Xlog:gc*:logs/gc.log:time,uptime:filecount=5,filesize=1M

GC logging does not hurt your performance and can be left on at all times. The files will not take up much space (5MB)

Technical Explanation of the Flags
===================================

#. **-Xms matching -Xmx – Why:** You should never run your server with the case that -Xmx can run the system completely out of memory. 
   Your server should always be expected to use the entire -Xmx! You should then ensure the OS has extra memory on top of that Xmx for non MC/OS level things. 
   Therefore, you should never run MC with -Xmx settings you can’t support if java uses it all. Now, that means if -Xms is lower than -Xmx 

>    **YOU HAVE UNUSED MEMORY! Unused memory is wasted memory.** G1 (and probably even CMS to a certain threshold, but we're only stating what we're sure about) 

Reviewer: CMS has been removed, not super relevant to mention here anymore (as well as other references)

   try to reclaim old gen memory faster to avoid running out of old.

#. **G1HeapRegionSize=8M+:** Default is auto calculated. SUPER important for Minecraft, especially 1.15, as 
   with low memory situations, the default calculation will in most times be too low. Any memory allocation half 
   of this size (4MB) will be treated as “Humongous” and promote straight to old generation and is harder to free. 
   If you allow java to use the default, you will be destroyed with a significant chunk of your memory getting treated as Humongous.

#. **+PerfDisableSharedMem:** Causes GC to write to file system which can cause major latency if disk IO is high 
   – See `<https://www.evanjones.ca/jvm-mmap-pause.html>`_

Using Large Pages
=================

Also for Large Pages – It’s even more important to use -Xms = -Xmx! Large Pages needs to have all of the memory specified 
for it or you could end up without the gains. This memory will not be used by the OS anyways, so use it.

Additionally use these flags (Metaspace is Java 8 Only, don’t use it for Java7): ``-XX:+UseLargePagesInMetaspace``

Transparent Huge Pages

> ======================

Reviewer: Inconsistent spacing throughout the document; keep one newline here and other places.
